Lots of warnings during build

# Bug

e.g.
```
main.c:400:8: warning: ‘free_v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  400 |     if (free_##u) {                 \
      |        ^
main.c:385:9: note: ‘free_v’ was declared here
  385 |     int free_##u = 0;               \
      |         ^~~~~
main.c:427:5: note: in expansion of macro ‘CHECK_OP’
  427 |     CHECK_OP(v, b);
      |     ^~~~~~~~
In file included from /usr/local/include/python3.14d/Python.h:69,
                 from pythoncapi_compat.h:21,
                 from main.c:1:
/usr/local/include/python3.14d/refcount.h:351:9: warning: ‘v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  351 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
main.c:427:14: note: ‘v’ was declared here
  427 |     CHECK_OP(v, b);
      |              ^
main.c:384:17: note: in definition of macro ‘CHECK_OP’
  384 |     MPZ_Object *u;                  \
      |                 ^
main.c: In function ‘add’:
main.c:400:8: warning: ‘free_v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  400 |     if (free_##u) {                 \
      |        ^
main.c:385:9: note: ‘free_v’ was declared here
  385 |     int free_##u = 0;               \
      |         ^~~~~
main.c:411:5: note: in expansion of macro ‘CHECK_OP’
  411 |     CHECK_OP(v, b);
      |     ^~~~~~~~
In file included from /usr/local/include/python3.14d/Python.h:69,
                 from pythoncapi_compat.h:21,
                 from main.c:1:
/usr/local/include/python3.14d/refcount.h:351:9: warning: ‘v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  351 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
main.c:411:14: note: ‘v’ was declared here
  411 |     CHECK_OP(v, b);
      |              ^
main.c:384:17: note: in definition of macro ‘CHECK_OP’
  384 |     MPZ_Object *u;                  \
      |                 ^
main.c: In function ‘mul’:
main.c:400:8: warning: ‘free_v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  400 |     if (free_##u) {                 \
      |        ^
main.c:385:9: note: ‘free_v’ was declared here
  385 |     int free_##u = 0;               \
      |         ^~~~~
main.c:470:5: note: in expansion of macro ‘CHECK_OP’
  470 |     CHECK_OP(v, b);
      |     ^~~~~~~~
In file included from /usr/local/include/python3.14d/Python.h:69,
                 from pythoncapi_compat.h:21,
                 from main.c:1:
/usr/local/include/python3.14d/refcount.h:351:9: warning: ‘v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  351 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
main.c:470:14: note: ‘v’ was declared here
  470 |     CHECK_OP(v, b);
      |              ^
main.c:384:17: note: in definition of macro ‘CHECK_OP’
  384 |     MPZ_Object *u;                  \
      |                 ^
main.c: In function ‘floordiv’:
main.c:400:8: warning: ‘free_v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  400 |     if (free_##u) {                 \
      |        ^
main.c:385:9: note: ‘free_v’ was declared here
  385 |     int free_##u = 0;               \
      |         ^~~~~
main.c:593:5: note: in expansion of macro ‘CHECK_OP’
  593 |     CHECK_OP(v, b);
      |     ^~~~~~~~
In file included from /usr/local/include/python3.14d/Python.h:69,
                 from pythoncapi_compat.h:21,
                 from main.c:1:
/usr/local/include/python3.14d/refcount.h:351:9: warning: ‘v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  351 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
main.c:593:14: note: ‘v’ was declared here
  593 |     CHECK_OP(v, b);
      |              ^
main.c:384:17: note: in definition of macro ‘CHECK_OP’
  384 |     MPZ_Object *u;                  \
      |                 ^
main.c: In function ‘divmod’:
main.c:400:8: warning: ‘free_v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  400 |     if (free_##u) {                 \
      |        ^
main.c:385:9: note: ‘free_v’ was declared here
  385 |     int free_##u = 0;               \
      |         ^~~~~
main.c:569:5: note: in expansion of macro ‘CHECK_OP’
  569 |     CHECK_OP(v, b);
      |     ^~~~~~~~
In file included from /usr/local/include/python3.14d/Python.h:69,
                 from pythoncapi_compat.h:21,
                 from main.c:1:
/usr/local/include/python3.14d/refcount.h:351:9: warning: ‘v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  351 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
main.c:569:14: note: ‘v’ was declared here
  569 |     CHECK_OP(v, b);
      |              ^
main.c:384:17: note: in definition of macro ‘CHECK_OP’
  384 |     MPZ_Object *u;                  \
      |                 ^
main.c: In function ‘bitwise_or’:
main.c:400:8: warning: ‘free_v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  400 |     if (free_##u) {                 \
      |        ^
main.c:385:9: note: ‘free_v’ was declared here
  385 |     int free_##u = 0;               \
      |         ^~~~~
main.c:918:5: note: in expansion of macro ‘CHECK_OP’
  918 |     CHECK_OP(v, b);
      |     ^~~~~~~~
In file included from /usr/local/include/python3.14d/Python.h:69,
                 from pythoncapi_compat.h:21,
                 from main.c:1:
/usr/local/include/python3.14d/refcount.h:351:9: warning: ‘v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  351 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
main.c:918:14: note: ‘v’ was declared here
  918 |     CHECK_OP(v, b);
      |              ^
main.c:384:17: note: in definition of macro ‘CHECK_OP’
  384 |     MPZ_Object *u;                  \
      |                 ^
main.c: In function ‘bitwise_xor’:
main.c:400:8: warning: ‘free_v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  400 |     if (free_##u) {                 \
      |        ^
main.c:385:9: note: ‘free_v’ was declared here
  385 |     int free_##u = 0;               \
      |         ^~~~~
main.c:1049:5: note: in expansion of macro ‘CHECK_OP’
 1049 |     CHECK_OP(v, b);
      |     ^~~~~~~~
In file included from /usr/local/include/python3.14d/Python.h:69,
                 from pythoncapi_compat.h:21,
                 from main.c:1:
/usr/local/include/python3.14d/refcount.h:351:9: warning: ‘v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  351 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
main.c:1049:14: note: ‘v’ was declared here
 1049 |     CHECK_OP(v, b);
      |              ^
main.c:384:17: note: in definition of macro ‘CHECK_OP’
  384 |     MPZ_Object *u;                  \
      |                 ^
main.c: In function ‘bitwise_and’:
main.c:400:8: warning: ‘free_v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  400 |     if (free_##u) {                 \
      |        ^
main.c:385:9: note: ‘free_v’ was declared here
  385 |     int free_##u = 0;               \
      |         ^~~~~
main.c:788:5: note: in expansion of macro ‘CHECK_OP’
  788 |     CHECK_OP(v, b);
      |     ^~~~~~~~
In file included from /usr/local/include/python3.14d/Python.h:69,
                 from pythoncapi_compat.h:21,
                 from main.c:1:
/usr/local/include/python3.14d/refcount.h:351:9: warning: ‘v’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  351 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
main.c:788:14: note: ‘v’ was declared here
  788 |     CHECK_OP(v, b);
      |              ^
main.c:384:17: note: in definition of macro ‘CHECK_OP’
  384 |     MPZ_Object *u;                  \
      |                 ^
```
They come from this
```
    #define CHECK_OP(u, a)              \
        MPZ_Object *u;                  \
        int free_##u = 0;               \
        if (MPZ_CheckExact(a)) {        \
            u = (MPZ_Object*)a;         \
        }                               \
        else if (PyLong_Check(a)) {     \
            u = from_int(a);            \
            if (!u) {                   \
                goto end;               \
            }                           \
            free_##u = 1;               \
        }                               \
        else {                          \
            Py_RETURN_NOTIMPLEMENTED;   \
        }
    #define FREE_OP(u)                  \
        if (free_##u) {                 \
            Py_DECREF(u);               \
        }
```

## environment
`Linux iZwz93d1vi9srmaw63ouuxZ 5.15.0-86-generic #96-Ubuntu SMP Wed Sep 20 08:23:49 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`

`gcc version 11.4.0 (Ubuntu 11.4.0-1ubuntu1~22.04)`

--- a/.github/workflows/basic.yml
+++ b/.github/workflows/basic.yml
@@ -8,6 +8,8 @@ jobs:
         python-version: [3.9, pypy3.9, '3.10', 'pypy3.10', 3.11, 3.12, 3.13]
         os: [ubuntu-22.04]
     runs-on: ${{ matrix.os }}
+    env:
+      CFLAGS: -Wpedantic -Werror
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5

--- a/main.c
+++ b/main.c
@@ -378,24 +378,19 @@ MPZ_add(MPZ_Object *u, MPZ_Object *v, int subtract)
 
 
 #define CHECK_OP(u, a)              \
-    MPZ_Object *u;                  \
-    int free_##u = 0;               \
+    static MPZ_Object *u;           \
     if (MPZ_CheckExact(a)) {        \
         u = (MPZ_Object*)a;         \
+        Py_INCREF(u);               \
     }                               \
     else if (PyLong_Check(a)) {     \
         u = from_int(a);            \
         if (!u) {                   \
             goto end;               \
         }                           \
-        free_##u = 1;               \
     }                               \
     else {                          \
         Py_RETURN_NOTIMPLEMENTED;   \
-    }
-#define FREE_OP(u)                  \
-    if (free_##u) {                 \
-        Py_DECREF(u);               \
     }
 
 
@@ -409,8 +404,8 @@ add(PyObject* a, PyObject *b)
 
     res = MPZ_add(u, v, 0);
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return res;
 }
 
@@ -425,8 +420,8 @@ sub(PyObject* a, PyObject *b)
 
     res = MPZ_add(u, v, 1);
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return res;
 }
 
@@ -468,8 +463,8 @@ mul(PyObject* a, PyObject *b)
 
     res = MPZ_mul(u, v);
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return (PyObject*)res;
 }
 
@@ -575,8 +570,8 @@ divmod(PyObject *a, PyObject *b)
     return res;
 end:
     Py_DECREF(res);
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return NULL;
 }
 
@@ -594,8 +589,8 @@ floordiv(PyObject *a, PyObject *b)
     Py_DECREF(r);
     return (PyObject*)q;
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return NULL;
 }
 
@@ -621,8 +616,8 @@ rem(PyObject *a, PyObject *b)
     Py_DECREF(q);
     return (PyObject*)r;
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return NULL;
 }
 
@@ -786,8 +781,8 @@ bitwise_and(PyObject *a, PyObject *b)
 
     res = (PyObject*)MPZ_and(u, v);
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return (PyObject*)res;
 }
 
@@ -916,8 +911,8 @@ bitwise_or(PyObject *a, PyObject *b)
 
     res = (PyObject*)MPZ_or(u, v);
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return (PyObject*)res;
 }
 
@@ -1047,8 +1042,8 @@ bitwise_xor(PyObject *a, PyObject *b)
 
     res = (PyObject*)MPZ_xor(u, v);
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return (PyObject*)res;
 }
 
@@ -1220,8 +1215,8 @@ richcompare(PyObject *a, PyObject *b, int op)
 
     int r = MPZ_Compare(u, v);
 
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     switch(op) {
     case Py_LT:
         return PyBool_FromLong(r == -1);
@@ -1236,10 +1231,10 @@ richcompare(PyObject *a, PyObject *b, int op)
     case Py_NE:
         return PyBool_FromLong(r != 0);
     }
-	Py_RETURN_NOTIMPLEMENTED;
+    Py_RETURN_NOTIMPLEMENTED;
 end:
-    FREE_OP(u);
-    FREE_OP(v);
+    Py_XDECREF(u);
+    Py_XDECREF(v);
     return NULL;
 }
 

